app/jobs/hyku_addons/reindex_model_job.rb

```diff
@@ -1,43 +1,38 @@
 # frozen_string_literal: true
+
+# call with
+#  use_work_ids an array is just for testing a few ids
+# HykuAddons::ReindexModelJob.perform_now("GenericWork", "repo.hyku.docker", limit: 1, options: { cname_doi_mint: ["repo.hyku.docker"], use_work_ids: ["15b3c16b-8183-4d2a-86ef-d144176fa0c8"]} )
+#
+# HykuAddons::ReindexAvailableWorksJob.perform_later(["repo.hyku.docker"], options: { cname_doi_mint: ["repo.hyku.docker"]} )
+
 module HykuAddons
   class ReindexModelJob < ApplicationJob
-    queue_as :reindex
-
     rescue_from Hyrax::DOI::DataCiteClient::Error, Ldp::Gone, Ldp::HttpError, RSolr::Error::Http, RSolr::Error::ConnectionRefused do |exception|
       Rails.logger.debug exception.inspect
       retry_job(wait: 5.minutes)
     end
 
-    def perform(klass, cname, limit: 25, page: 1, cname_doi_mint: [])
+    def perform(klass, cname, limit: 25, page: 1, options: {})
+      options ||= options || { cname_doi_mint: [], use_work_ids: [] }
       # for whatever in private methods reason without assigning it to instamce variable it throws undefined local variable
-      @cname_doi_mint = cname_doi_mint
+      @cname_doi_mint = options[:cname_doi_mint]
+      @use_work_ids = options[:use_work_ids]
       @cname = cname
+      @limit = limit
+      @page = page
+      @klass = klass
       AccountElevator.switch!(cname)
 
-      Rails.logger.debug "=== Starting to reindex #{klass} in #{cname} ==="
-
-      offset = (page - 1) * limit
-
-      # When the offset becomes too large, no records would be found
-      works =  klass.constantize.where("title_tesim:*").limit(limit).offset(offset)
-
-      # with calling to_a it always returns true, even when no records found
-      if works.to_a.any?
-        reindex_works(works)
-
-        new_page_count = page.to_i + 1
-
-        # Re-enqueue
-        ReindexModelJob.perform_later(klass, cname, page: new_page_count)
-      end
-
-      Rails.logger.debug "=== Completed reindex of #{klass} in #{cname} ==="
+      offset = (page - 1) * @limit
+      work_class = klass.constantize
+      fetch_work_using_ids(work_class) if @use_work_ids.present?
```

Comment:
Can we use a more immediately apparent condition?

```
if @use_work_ids.present?
    fetch_work_using_ids(work_class) 
else
    reindex_and_mint_work(work_class, offset) 
end
```

Reply:
I thought you loved saving lines, was done with you in mind.

I will change it though/

Reply:
hahaha. I do like saving lines, but clarity comes first  :D

I think its just this pattern that Hyrax uses, with these kind of end of line conditions for if/else... IMO they are not easier to read.